schema_salad/python_codegen.py

```diff
@@ -581,9 +605,16 @@ def declare_field(
                 fmt(
                     """
 if self.{safename} is not None:
-    r["{fieldname}"] = save(
-        self.{safename}, top=False, base_url={baseurl}, relative_uris=relative_uris
+    saved_val = save(
+        self.{safename}, top=False, base_url={baseurl}, relative_uris=relative_uris, keys = keys + ["{fieldname}"]
     )
+
+    if type(saved_val) == list:
+        if len(saved_val) == 1: # If the returned value is a list of size 1, just save the value in the list
+            saved_val = saved_val[0]
+    r["{fieldname}"] = saved_val
+
+    max_len = add_kv(old_doc = doc, new_doc = r, line_numbers = line_numbers, key = "{fieldname}", val = r.get("{fieldname}"), max_len = max_len, cols = cols)
```

Comment:
Since you have `r`, `key` and `val` together already, I would consider moving the assignment of `r["{fieldname}"] = saved_val` into `add_kv()` so the `add_kv()` method is responsible for setting both the value and metadata of the entry at the same time.